types/LinkExitTypes.d.ts

```diff
@@ -0,0 +1,157 @@
+import { LinkInstitution } from "./LinkInstitution";
+
+export interface LinkExit {
+    error?: LinkError;
+    metadata: LinkExitMetadata;
+}
+
+export interface LinkExitMetadata {
+    link_session_id: string;
+    request_id: string;
+    status?: LinkExitMetadataStatus;
+    institution?: LinkInstitution;
+    metadataJson?: String;
+}
+
+export enum LinkExitMetadataStatus {
+    CONNECTED = 'connected',
+    CHOOSE_DEVICE = 'choose_device',
+    REQUIRES_CODE = 'requires_code',
+    REQUIRES_CREDENTIALS = 'requires_credentials',
+    REQUIRES_EXTERNAL_ACTION = 'requires_external_action',
+    REQUIRES_OAUTH = 'requires_oauth',
+    REQUIRES_QUESTIONS = 'requires_questions',
+    REQUIRES_RECAPTCHA = 'requires_recaptcha',
+    REQUIRES_SELECTIONS = 'requires_selections',
+    REQUIRES_DEPOSIT_SWITCH_ALLOCATION_CONFIGURATION = 'requires_deposit_switch_allocation_configuration',
+    REQUIRES_DEPOSIT_SWITCH_ALLOCATION_SELECTION = 'requires_deposit_switch_allocation_selection',
+}
+
+export interface LinkError {
+    error_code: LinkErrorCode;
+    error_type: LinkErrorType;
```

Comment:
iOS and Android don't have an errorType (For them it's a subtype of ErrorCode)

Reply:
I talked with swb and nesting types doesn't make as much sense in ts so we decided to break it back out.  This means the bridge code for Android and iOS will have to break them apart